src/decoder/plugins/UsfDecoderPlugin.cxx

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright (C) 2003-2015 The Music Player Daemon Project
+ * http://www.musicpd.org
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "config.h"
+#include "UsfDecoderPlugin.hxx"
+#include "../DecoderAPI.hxx"
+#include "../DecoderInternal.hxx"
```

Comment:
You did it the wrong way: you're the decoder, you should know the timestamp, but you didn't tell the MPD core via `decoder_timestamp()`. In spite of that, you ask the MPD core for the timestamp, which it can only guess by counting samples.

It is up to you (as the decoder) to know the time stamp. There can be various ways to keep track of that; some codecs contain a time stamp in every packet/block, and others need to count samples.